### PR TITLE
Add contrail test

### PR DESCRIPTION
**Pull request by eonpatapon**

### CODE REVIEW

File: test/contrail.nix [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+      src = pkgs.fetchzip {
+        url = "https://git.corp.cloudwatt.com/applications/deployment/repository/${rev}/archive.tar.gz";
+        inherit sha256;
+      };
+      phases = [ "unpackPhase" "installPhase" "fixupPhase" ];
+      installPhase = ''
+        mkdir -p $out
+        cp -r * $out/
+      '';
+      postFixup = ''
```
Comment (nlewo) [inline test/contrail.nix]: would be better to this in the patchPhase

File: test/contrail.nix [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+      src = pkgs.fetchzip {
+        url = "https://git.corp.cloudwatt.com/docker/keystone-all/repository/${rev}/archive.tar.gz";
+        inherit sha256;
+      };
+      phases = [ "unpackPhase" "installPhase" "fixupPhase" ];
+      installPhase = ''
+        mkdir -p $out
+        cp -r * $out/
+      '';
+      postFixup = ''
```
Comment (nlewo) [inline test/contrail.nix]: do it in the patch phase

File: pkgs/lib/images.nix [hunk 3]
```diff
@@ -21,9 +21,9 @@
   };
 
   developmentDnsmaskImage = pkgs.dockerTools.pullImage {
-    imageName = "r.cwpriv.net/development/dnsmasq";
-    imageTag = "2.75-024c101a4eb107e6";
-    sha256 = "1rg6mnf6fwjqgh15fqf31rqxh4ggwb5cl5pd72d83z41yiv5ws46";
+    imageName = "r.cwpriv.net/eon/dnsmasq";
```
Comment (nlewo) [inline pkgs/lib/images.nix]: Could we integrate this upstream ?

Comment (eonpatapon) [reply pkgs/lib/images.nix]: https://git.corp.cloudwatt.com/docker/development-dnsmasq/merge_requests/2

File: test/contrail.nix (hunk 2 above)
Comment (eonpatapon) [reply test/contrail.nix]: done

File: test/contrail.nix (hunk 1 above)
Comment (eonpatapon) [reply test/contrail.nix]: done

**Pull request merged**